src/state_machine.rs

```diff
@@ -245,35 +266,33 @@ impl Debug for StateMachine {
     }
 }
 
-#[cfg(not(feature = "use-mock-crust"))]
 impl StateMachine {
-    /// Run the event loop for sending and receiving messages. Blocks until
-    /// the core is terminated, so it must be called in a separate thread.
-    pub fn run(&mut self) {
-        while self.is_running {
+    #[cfg(not(feature = "use-mock-crust"))]
```

Comment:
It's unreachable (and considered unused) when compiling with mock crust at the moment. We could always remove the `cfg` if it turns out to be useful in a test?